packages/block-library/src/file/edit.native.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import { View, Text } from 'react-native';
+import React from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockIcon,
+	MediaPlaceholder,
+	MediaUploadProgress,
+	RichText,
+	PlainText,
+} from '@wordpress/block-editor';
+import { file as icon } from '@wordpress/icons';
+import { Component } from '@wordpress/element';
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+
+export default class FileEdit extends Component {
```

Comment:
@etoledom :) not sure if this was asked elsewhere but I am just checking on the reason we are using a class component instead of a function component since function components lessen the amount of code in a component primarily with the use of hooks.

Reply:
This was my starting point as I based it on another block. But I agree we should have this as a function component.
I had this planned after merging the second round of PR (that was when I noticed we should do this). Luckily the rewrites to functions are not complex.